src/deluge/io/midi/midi_follow.cpp

```diff
@@ -347,43 +347,40 @@ PLACE_SDRAM_DATA Clip* clipForLastNoteReceived[kMaxMIDIValue + 1] = {0};
 /// and should be routed to the active context for further processing
 void MidiFollow::noteMessageReceived(MIDIDevice* fromDevice, bool on, int32_t channel, int32_t note, int32_t velocity,
                                      bool* doingMidiThru, bool shouldRecordNotesNowNow, ModelStack* modelStack) {
-	// midi follow mode
-	if (midiEngine.midiFollow) {
-		MIDIMatchType matchSynth = checkMidiFollowMatch(fromDevice, channel, MIDIFollowChannelType::SYNTH);
-		MIDIMatchType matchKit = checkMidiFollowMatch(fromDevice, channel, MIDIFollowChannelType::KIT);
-		if ((matchSynth != MIDIMatchType::NO_MATCH) || (matchKit != MIDIMatchType::NO_MATCH)) {
-			Clip* clip;
-			if (on) {
-				clip = getClipForMidiFollow(true);
-			}
-			else {
-				// for note off's, see if a note on message was previously sent so you can
-				// direct the note off to the right place
-				clip = clipForLastNoteReceived[note];
-			}
+	MIDIMatchType matchSynth = checkMidiFollowMatch(fromDevice, channel, MIDIFollowChannelType::SYNTH);
+	MIDIMatchType matchKit = checkMidiFollowMatch(fromDevice, channel, MIDIFollowChannelType::KIT);
```

Comment:
I think the logic in here can be improved - most of the time both channels will be the same, and when they're not just sending the note to midi follow in both cases makes more sense than dropping it. I'd suggest updating your check match function to just return a single match type

Alternatively if you tracked the last kit and last synth separately this method would make sense